internal/fpath/temp_dir.go

```diff
@@ -0,0 +1,26 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+// TODO maybe there is better place for this
+
+package fpath
+
+import "context"
+
+// The key type is unexported to prevent collisions with context keys defined in
+// other packages.
+type key int
+
+// tempDir is the context key for temporary directory
+const tempDir key = 0
+
+// WithTempDir creates context with api key
+func WithTempDir(ctx context.Context, dir string) context.Context {
```

Comment:
instead of using `inmemory` as a special directory name that means "don't use the directory called `inmemory` as a directory, actually store data in memory", perhaps we should store more than just a string with the context key? if we stored a struct or another value, we could keep track of the inmemory/ondisk distinction separately from the directory name